From cff0a7522a372f4b8c466312eaaa62cd6a762f2e Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Fri, 2 Dec 2016 11:46:25 +0100
Subject: [PATCH 06/13] afc: Leak mutex when force unmounting

The afc daemon sometimes aborts on g_mutex_clear when force unmounting.
Leak the mutex in order to avoid the aborts. The daemon will hopefully
finish successfully then...

https://bugzilla.gnome.org/show_bug.cgi?id=775514
---
 daemon/gvfsbackendafc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/daemon/gvfsbackendafc.c b/daemon/gvfsbackendafc.c
index 00936b15..befb3a9e 100644
--- a/daemon/gvfsbackendafc.c
+++ b/daemon/gvfsbackendafc.c
@@ -183,7 +183,8 @@ g_vfs_backend_afc_close_connection (GVfsBackendAfc *self)
               sbservices_client_free (self->sbs);
               self->sbs = NULL;
             }
-          g_mutex_clear (&self->apps_lock);
+          if (self->force_umount_id == 0)
+            g_mutex_clear (&self->apps_lock);
         }
       else
         {
-- 
2.15.1

